fpmsyncd/routesync.cpp

```diff
@@ -2393,3 +2601,141 @@ void RouteSync::onWarmStartEnd(DBConnector& applStateDb)
         SWSS_LOG_NOTICE("Warm-Restart reconciliation processed.");
     }
 }
+
+/*
+ * Get nexthop group key as string
+ * @arg id     next hop group id
+ *
+ * Return nexthop group key
+ */
+const string RouteSync::getNextHopGroupKeyAsString(uint32_t id) const
+{
+    return string("ID") + to_string(id);
+}
+
+/*
+ * update the nexthop group entry
+ * @arg nh_id     nexthop group id
+ *
+ */
+void RouteSync::updateNextHopGroup(uint32_t nh_id)
+{
+    auto git = m_nh_groups.find(nh_id);
+    if(git == m_nh_groups.end())
+    {
+        SWSS_LOG_ERROR("Nexthop not found: %d", nh_id);
+        return;
+    }
+
+    NextHopGroup& nhg = git->second;
+
+    if(nhg.installed)
+    {
+        //Nexthop group already installed
+        return;
+    }
+    nhg.installed = true;
+    updateNextHopGroupDb(nhg);
+}
+
+/*
+ * delete the nexthop group entry
+ * @arg nh_id     nexthop group id
+ *
+ */
+void RouteSync::deleteNextHopGroup(uint32_t nh_id)
+{
+    auto git = m_nh_groups.find(nh_id);
+    if(git == m_nh_groups.end())
+    {
+        SWSS_LOG_ERROR("Nexthop not found: %d", nh_id);
+        return;
+    }
+
+    NextHopGroup& nhg = git->second;
+
+    if(nhg.installed)
+    {
+        string key = getNextHopGroupKeyAsString(nh_id);
+        m_nexthop_groupTable.del(key.c_str());
+        SWSS_LOG_DEBUG("NextHopGroup table del: key [%s]", key.c_str());
+    }
+    m_nh_groups.erase(git);
+}
+
+/*
+ * update the nexthop group table in database
+ * @arg nhg     the nexthop group
+ *
+ */
+void RouteSync::updateNextHopGroupDb(const NextHopGroup& nhg)
+{
+    vector<FieldValueTuple> fvVector;
+    string nexthops;
+    string ifnames;
+    string weights;
+    string key = getNextHopGroupKeyAsString(nhg.id);
+    getNextHopGroupFields(nhg, nexthops, ifnames, weights);
+
+    FieldValueTuple nh("nexthop", nexthops.c_str());
+    FieldValueTuple ifname("ifname", ifnames.c_str());
+    fvVector.push_back(nh);
+    fvVector.push_back(ifname);
+    if(!weights.empty())
+    {
+        FieldValueTuple wg("weight", weights.c_str());
+        fvVector.push_back(wg);
+    }
+    SWSS_LOG_INFO("NextHopGroup table set: key [%s] nexthop[%s] ifname[%s] weight[%s]", key.c_str(), nexthops.c_str(), ifnames.c_str(), weights.c_str());
+
+    //TODO: Take care of warm reboot
```

Comment:
Please don't add TODOs in the code. Warmboot is an important scenario. This need to be handled.

Reply:
what is this TODO for? What warm reboot case do you plan to handle here?

Reply:
Removed TODO from the code.

Reply:
We have talked before in RoutingWG about warm reboot needing FRR support.